File: app/MoviesApp/app/src/main/java/com/example/moviesapp/view/screens/componentslist/ComponentsFragment.java
package com.example.moviesapp.view.screens.componentslist;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.example.moviesapp.databinding.FragmentComponentsListBinding;
import com.example.moviesapp.view.BaseFragment;
import com.example.moviesapp.view.ScreenView;

/**
 * Components fragment.
 */
public class ComponentsFragment extends BaseFragment<FragmentComponentsListBinding> {
    @Override
    protected FragmentComponentsListBinding binding(LayoutInflater inflater, ViewGroup container) {
        return FragmentComponentsListBinding.inflate(
                inflater,
                container,
                false
        );
    }

    @Override
    protected ScreenView<FragmentComponentsListBinding> view() {
        return new ComponentsView();
    }
}

File: app/kotlin/MoviesApp/app/src/main/java/com/example/moviesapp/view/screens/componentslist/ComponentsFragment.kt
package com.example.moviesapp.view.screens.componentslist

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import com.example.moviesapp.data.movies.model.Movie
import com.example.moviesapp.databinding.FragmentComponentsListBinding
import com.example.moviesapp.navigation.NavigationFragment
import com.example.moviesapp.view.screens.moviescreen.impl.MovieDetailsScreenFragment

/**
 * Components fragment.
 */
internal class ComponentsFragment: NavigationFragment<FragmentComponentsListBinding>() {

    private val componentsNavigator: (Movie) -> Unit =
        {
            val bundle = Bundle()
            bundle.putString("id", it.id)
            val movieDetailsScreenFragment = MovieDetailsScreenFragment()
            navigateTo(
                    bundle,
                    movieDetailsScreenFragment
            )
        }

    override fun binding(inflater: LayoutInflater, container: ViewGroup) =
            FragmentComponentsListBinding.inflate(
                    inflater,
                    container,
                    false
            )

    override fun view() = ComponentsView(componentsNavigator)
}